Neos.Flow/Classes/Reflection/ReflectionService.php

```diff
@@ -1727,29 +1825,15 @@ protected function convertParameterReflectionToArray(ParameterReflection $parame
      * accordingly.
      *
      * @return void
+     * @throws ClassLoadingForReflectionFailedException
+     * @throws InvalidClassException
+     * @throws ReflectionException
      */
     protected function forgetChangedClasses(): void
     {
-        $frozenNamespaces = [];
-        foreach ($this->packageManager->getAvailablePackages() as $packageKey => $package) {
-            if ($this->packageManager->isPackageFrozen($packageKey)) {
-                $frozenNamespaces = array_merge($frozenNamespaces, $package->getNamespaces());
-            }
-        }
-        $frozenNamespaces = array_unique($frozenNamespaces);
-
         $classNames = array_keys($this->classReflectionData);
-        foreach ($frozenNamespaces as $namespace) {
-            $namespace .= '\\';
-            foreach ($classNames as $index => $className) {
-                if (str_starts_with($className, $namespace)) {
-                    unset($classNames[$index]);
-                }
-            }
-        }
-
         foreach ($classNames as $className) {
-            if (!$this->statusCache->has($this->produceCacheIdentifierFromClassName($className))) {
+            if (is_string($className) && !$this->reflectionDataRuntimeCache->has($this->produceCacheIdentifierFromClassName($className))) {
```

Comment:
What – if not a string – can `$className` be here?

Reply:
I had an example were that was _for some reason I couldn't figure out_ a bool, so I jjust wanted to be sure here....

Reply:
i think that must have been in a previous iteration of the code as `array_keys` is used which can NEVER return a bool as that is not a valid array key.